lib/codebreaker/statistics.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Codebreaker
+  class Statistics
+    include Constants::Shared
+
+    PERMITTED_CLASSES = [Codebreaker::User, Symbol].freeze
+    YAML_OPTIONS = { permitted_classes: PERMITTED_CLASSES, aliases: true }.freeze
+
+    def initialize(filename = 'default.yml')
+      @filename = filename
+      @data_store = []
+    end
+
+    def add_to_data_store(game)
+      @data_store.push(data_to_save(game))
+    end
+
+    def save_unit
+      @data_store.empty? ? raise(StandardError, I18n.t(:empty_data)) : File.write(@filename, @data_store.to_yaml)
+    end
+
+    def show
+      validate_file? ? sort_store(YAML.load_file(@filename, **YAML_OPTIONS)) : raise(StandardError, I18n.t(:bad_file))
+    end
+
+    private
+
+    def sort_store(data)
```

Comment:
add specs to show that sorting is correct

Reply:
need help